src/components/Reports/Reports.js

```diff
@@ -50,6 +52,28 @@ const Reports = ({ dependencies: { datahubClient } }) => {
     }));
   };
 
+  useEffect(() => {
+    const fetchVisitsByPeriod = async () => {
+      const visitsWithoutEmail = await datahubClient.getTotalVisitsOfPeriod({
+        domainName: state.domainSelected.name,
+        dateFrom: state.dateFrom,
+        dateTo: state.dateTo,
+        emailFilter: 'without_email',
+      });
+      const visitsWithEmail = await datahubClient.getTotalVisitsOfPeriod({
+        domainName: state.domainSelected.name,
+        dateFrom: state.dateFrom,
+        dateTo: state.dateTo,
+        emailFilter: 'with_email',
+      });
+
+      setTotalVisits({ withEmail: visitsWithEmail, withoutEmail: visitsWithoutEmail });
+    };
+    if (state.domainSelected) {
+      fetchVisitsByPeriod();
+    }
+  }, [datahubClient, state.domainSelected, state.dateFrom, state.dateTo]);
+
```

Comment:
Oh, you are keeping the refactorized old reports page. 

I imagined starting to splitting both pages and controls here. 

If you are comfortable with this approach, go ahead, if you feel that it is difficult, let me know and I can share with you what I was imagining.